hardware_interface/src/component_parser.cpp

```diff
@@ -103,6 +107,61 @@ std::string get_attribute_value(
   return get_attribute_value(element_it, attribute_name, std::string(tag_name));
 }
 
+/// Parse optional size attribute
+/**
+ * Parses an XMLElement and returns the value of the size attribute.
+ * If not specified, defaults to 1. If not given a positive int, throws an error.
+ *
+ * \param[in] elem XMLElement that has the size attribute.
+ * \return The size.
+ * \throws std::runtime_error if not given a positive non-zero int as value.
```

Comment:
Very much nit-picking :P

```suggestion
 * If not specified, defaults to 1. If not given a positive integer, throws an error.
 *
 * \param[in] elem XMLElement that has the size attribute.
 * \return The size.
 * \throws std::runtime_error if not given a positive non-zero integer as value.
```